### PR TITLE
camerad:  centralize enabled handling and simplify camera Initialization  

### PR DESCRIPTION
**Pull request by deanlee**
1. **Centralized enabled Handling:** Removed redundant enabled checks from individual functions. The enabled state is now handled during the camera initialization phase, ensuring that cameras are set up correctly at the start.
2. **Simplified Camera Initialization:** Moved all camera setup tasks to `CameraState::init`. This consolidates the initialization process, making the flow more streamlined and easier to maintain.
3. **Simplified Thread Management:** Threads are now managed directly by `CameraState`, removing the need to track them in a separate vector or manually join them on exit.

### CODE REVIEW

File: system/camerad/cameras/spectra.cc [hunk 1]
```diff
@@ -172,7 +165,6 @@ void SpectraCamera::sensors_i2c(const struct i2c_random_wr_payload* dat, int len
   int ret = device_config(sensor_fd, session_handle, sensor_dev_handle, cam_packet_handle);
   if (ret != 0) {
     LOGE("** sensor %d FAILED i2c, disabling", cc.camera_num);
-    enabled = false;
```
Comment (adeebshihadeh) [inline system/camerad/cameras/spectra.cc:175]: the camera will stay running now?

File: system/camerad/cameras/spectra.cc [hunk 2]
```diff
@@ -516,10 +507,6 @@ bool SpectraCamera::openSensor() {
 }
 
 void SpectraCamera::configISP() {
-  // NOTE: to be able to disable road and wide road, we still have to configure the sensor over i2c
-  // If you don't do this, the strobe GPIO is an output (even in reset it seems!)
```
Comment (adeebshihadeh) [inline system/camerad/cameras/spectra.cc:520]: let's keep this and put it in `tici.h`

File: system/camerad/cameras/camera_qcom2.cc [hunk 3]
```diff
@@ -397,6 +403,4 @@ void camerad_thread() {
   }
 
   LOG(" ************** STOPPING **************");
```
Comment (adeebshihadeh) [inline system/camerad/cameras/camera_qcom2.cc:405]: remove this

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: system/camerad/cameras/spectra.cc (hunk 1 above)
Comment (deanlee) [reply system/camerad/cameras/spectra.cc:175]: Yes, should we  add an assert here ? After this error, there's no reason to keep camerad running, so a restart may be the best option.

Comment (adeebshihadeh) [reply system/camerad/cameras/spectra.cc:175]: `camerad` should always output as many frames as it can

**Comment by deanlee** [comment]
It should be fine now. still keep the `enabled` variable in `CameraState` solely to disable the camera if sensors_i2c fails, allowing `handle_camera_event` to skip processing for that camera. The use of enabled for initialization masking during startup in other parts of the original code can be safely removed.

**Pull request closed without merging**